execution_engine/src/storage/global_state/lmdb.rs

```diff
@@ -281,16 +289,58 @@ impl StateProvider for LmdbGlobalState {
         correlation_id: CorrelationId,
         trie_keys: Vec<Digest>,
     ) -> Result<Vec<Digest>, Self::Error> {
-        let txn = self.environment.create_read_txn()?;
-        let missing_descendants =
-            missing_trie_keys::<Key, StoredValue, lmdb::RoTransaction, LmdbTrieStore, Self::Error>(
+        if trie_keys
+            .iter()
+            .filter(|digest| {
+                !self
+                    .digests_without_missing_descendants
+                    .read()
```

Comment:
Would it be cheaper to acquire the read lock only once, before filtering, instead of in every iteration?

Reply:
Yup, Dan pointed this one too, thanks!
Fixed in aac1d6b4d9dba612378a1af8429fe70f09415908